include/openmc/distribution.h

```diff
@@ -60,7 +58,32 @@ class Discrete : public Distribution {
   void normalize();
 
   //! Initialize alias tables for distribution
-  void init_alias(vector<double>& x, vector<double>& p);
+  void init_alias();
+};
+
+//==============================================================================
+//! A discrete distribution (probability mass function)
+//==============================================================================
+
+class Discrete : public Distribution {
+public:
+  explicit Discrete(pugi::xml_node node);
+  Discrete(const double* x, const double* p, int n);
+
+  //! Sample a value from the distribution
+  //! \param seed Pseudorandom number seed pointer
+  //! \return Sampled value
+  double sample(uint64_t* seed) const override;
+
+  // Properties
+  const vector<double>& x() const { return x_; }
+  const vector<double>& prob() const { return di_->prob(); }
+  const vector<size_t>& alias() const { return di_->alias(); }
+
+private:
+  vector<double> x_;             //!< Possible outcomes
+  unique_ptr<DiscreteIndex> di_; //!< discrete probability distribution of
```

Comment:
I would just hold a plain `DiscreteIndex` -- no particular need to make this a pointer.